whois-api/src/main/java/net/ripe/db/whois/api/rest/HttpsBasicAuthCustomizer.java

```diff
@@ -0,0 +1,90 @@
+package net.ripe.db.whois.api.rest;
+
+import com.google.common.net.HttpHeaders;
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletRequestWrapper;
+import jakarta.servlet.http.HttpServletResponse;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jetty.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.codec.binary.Base64;
+import static jakarta.servlet.http.HttpServletRequest.BASIC_AUTH;
+
+@Component
+public class HttpsBasicAuthCustomizer implements Filter {
+
+    @Override
+    public void doFilter(final ServletRequest request, final ServletResponse response, final FilterChain chain) throws IOException, ServletException {
+
+        if(!canProceed(request)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        final HttpServletRequest httpRequest = (HttpServletRequest) request;
+
+        if (RestServiceHelper.isHttpProtocol(httpRequest)){
+            final HttpServletResponse httpResponse = (HttpServletResponse)response;
+            httpResponse.sendError(HttpStatus.UPGRADE_REQUIRED_426, "HTTPS required for Basic authorization");
+            return;
+        }
+
+        chain.doFilter(new HttpBasicAuthRequestWrapper((HttpServletRequest) request), response);
+    }
+
+    private boolean canProceed(final ServletRequest request) {
+       if(! (request instanceof HttpServletRequest)) {
+           return false;
+       }
+
+       if(!RestServiceHelper.isBasicAuth((HttpServletRequest) request)) {
+            return false;
+       }
+
+       return true;
+    }
+
+    @Override
+    public void destroy() {
+        // do nothing
+    }
+
+    private static final class HttpBasicAuthRequestWrapper extends HttpServletRequestWrapper {
+
+        private HttpBasicAuthRequestWrapper(final HttpServletRequest request) {
+            super(request);
+        }
+
+        @Override
+        public String getQueryString() {
+            final String basicAuthPassword = getBasicAuthPassword(getHeader(HttpHeaders.AUTHORIZATION));
+            if(basicAuthPassword == null) {
+                return super.getQueryString();
+            }
+
+            final StringBuilder modifiedQueryString = (super.getQueryString() == null) ?
+                                                             new StringBuilder() :
+                                                             new StringBuilder(super.getQueryString()).append("&");
+
+            return modifiedQueryString.append("password=").append(basicAuthPassword).toString();
```

Comment:
will this code add password=xxxx in all the endpoints even if the endpoint do not have that parameter?

I mean if the user uses basicAuth in one endpoint that does not support "password" query parameter, we will add that parameter there, right?

This will be applied to all the API services

Reply:
Yes t will be applied to all services and Yes parameters are optional in Http request

Reply:
mmm...okok. Thanks for the answer!